lib/streamlit/caching.py

```diff
@@ -514,13 +529,13 @@ def get_or_set_cache():
             hasher = hashlib.new("md5")
 
             args_hasher = CodeHasher("md5", hasher)
-            args_hasher.update([args, kwargs])
+            args_hasher.update([args, kwargs], hash_funcs=hash_funcs)
```

Comment:
I think it would be cleaner to just pass `hash_funcs` to the `CodeHasher` constructor instead. Same elsewhere below.